vanessaspear/passport-api
# Created a stamps view to handle requests to /stamps

## Description 

A user is able to create stamps (photos, journals, and products).  The stamp models and view were created to handle this functionality.

## Changes

- Created a StampProduct model
- Created a StampPhoto model
- Created a StampJournal model
- Created a basic stamp view to handle requests to these models
- Added a photos custom action to handle requests to /stamps/photos

## Requests / Responses

**Request**

POST `/stamps/photos` Creates a new stamp photo

```
{
     description: "Sailboats are beautiful"
     image: "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAA
     itinerary: 2
     public: true
     trip: 1
     type: 1
}
```

**Response**

HTTP/1.1 201 OK

```json
{
     "id": 3,
     "image": "http://localhost:8000/media/photos/stampPhoto-18a0d62e-d870-48b6-8393- 
      763d5c81369a.jpeg",
     "description": "Sailboats are beautiful",
     "public":true,
     "trip":1,
     "itinerary":2,
     "type":1,
     "date_created":"2023-03-22T14:22:44.606433Z"
}
```

## Testing

```
git fetch origin vs-stamp-view
git checkout vs-stamp-view
```
- Pull down changes from passport-client repo, pull request # 
- Create and run database migrations 
- Navigate to stamps and click on the add stamp button 
- Select a trip from the dropdown and select photo from the type dropdown
- Upload a photo and add a caption 
- Click the save button 
- Verify the new stamp photo posted to the database
- Note: The photo won't show on the app at this time.  The stamps list component still needs to be created. 


## Related Issues

- Closes #20
